suncourse.c

```diff
@@ -0,0 +1,171 @@
+#include "dark_mode_switcher.h"
+
+#include <math.h>
+
+/*
+Based on SUNRISET.C
+Released to the public domain by Paul Schlyter, December 1992
+Source: https://stjarnhimlen.se/comp/sunriset.c
+*/
+
+typedef union {
+    FILETIME FtPart;
+    ULONGLONG QuadPart;
+} FILETIME_QUAD;
```

Comment:
I forgot to mention this little rascal.  
https://learn.microsoft.com/en-us/windows/win32/api/minwinbase/ns-minwinbase-filetime#remarks  
> Do not cast a pointer to a FILETIME structure to either a ULARGE_INTEGER* or __int64* value because it can cause alignment faults on 64-bit Windows.  

I'm afraid I just disguised the type punning via pointer by a type punning via union 🤣 Do you consider this an issue?